src/services/formatting/smartIndenter.ts

```diff
@@ -174,22 +179,24 @@ namespace ts.formatting {
                     indentationDelta += options.indentSize;
                 }
 
+                // Update `current` and `parent`.
+
+                const useTrueStart =
+                    isArgumentAndStartLineOverlapsExpressionBeingCalled(parent, current, currentStart.line, sourceFile);
+
                 current = parent;
-                currentStart = parentStart;
                 parent = current.parent;
+                currentStart = useTrueStart ? sourceFile.getLineAndCharacterOfPosition(current.getStart()) : containingListOrParentStart;
+                containingListOrParentStart = undefined;

```

Comment:
What's the point or setting this to undefined at the end? Looks like it's always reassigned inside the loop, and unused outside of it.

Reply:
It was placed to emphasize that the value of `containingListOrParentStart` is meaningless at the start of the loop, but this could be seen by just looking at what happens in the first iteration.

Reply:
Removed.